core-bundle/src/Resources/contao/modules/ModuleSearch.php

```diff
@@ -311,7 +311,7 @@ protected function addImageToTemplateFromSearchResult(array $result, Template $t
 			}
 
 			$figureBuilder = System::getContainer()->get('contao.image.studio')->createFigureBuilder();
-			$figureBuilder->fromPath($v['https://schema.org/primaryImageOfPage']['contentUrl']);
+			$figureBuilder->fromPath(ltrim($v['https://schema.org/primaryImageOfPage']['contentUrl'], '/'));
```

Comment:
```suggestion
			$figureBuilder->fromPath(urldecode(ltrim(parse_url($v['https://schema.org/primaryImageOfPage']['contentUrl'], PHP_URL_PATH), '/')));
```

The `contentUrl` is prefixed with the CDN URL if configured, so we need to make sure to only use the path of the URL. And because it is a URL, it needs to be decoded to be able to use it as a path.

How should this work if `contentUrl` points to a completely different domain that is not associated with the installation this code is running in?

Reply:
Maybe a new method `FigureBuilder::fromUrl()` would make sense here that handles all these cases?
/cc @m-vo

Reply:
See https://github.com/contao/contao/pull/5855

Reply:
Should we add the `figureBuilder->fromPath()` to `figureBuilder->fromUrl()` change in the `ModuleSearch` to #5855 too and close this PR? We'd then immediately know from the history why it was introduced.

Reply:
Done in [contao/contao@`3ab81ac` (#5855)](https://github.com/contao/contao/pull/5855/commits/3ab81ace056ef55be9caca8f362b397dab0df1ad)